llama-index-integrations/postprocessor/llama-index-postprocessor-voyageai-rerank/llama_index/postprocessor/voyageai_rerank/base.py

```diff
@@ -15,11 +15,13 @@
 
 class VoyageAIRerank(BaseNodePostprocessor):
     model: str = Field(description="Name of the model to use.")
-    top_n: int = Field(
-        description="The number of most relevant documents to return. If not specified, the reranking results of all documents will be returned."
+    top_n: Optional[int] = Field(
+        description="The number of most relevant documents to return. If not specified, the reranking results of all documents will be returned.",
+        default=None,
```

Comment:
This will break `ReRankStartEvent` if its None

Reply:
maybe in that case, we pass in -1 ?

Reply:
Ah true, my bad. We could pass `-1` or we could calculate `TOP_K` on the fly as `len(nodes)` to not have to dispatch a dummy value?

Reply:
Also should this have been caught in a unit test? Do I need to add another one?

Reply:
To add a proper unit test for this, you'd need to mock out the call to voyageai -- feel free to add this

Reply:
len(nodes) is a good option too

Reply:
I implemented the fix and edited one of the unit tests to make sure the events are dispatched properly.